pyccel/ast/literals.py

```diff
@@ -164,25 +157,22 @@ def python_value(self):
 class LiteralImaginaryUnit(LiteralComplex):
     """Represents the python value j"""
     def __new__(cls):
-        return LiteralComplex.__new__(cls, 0, 1)
```

Comment:
Do we still need to override the `__new__` constructor, given that we have an `__init__` function?

Reply:
Yes. `LiteralComplex` has an explicit `__new__` function which converts `LiteralComplex(0,1)` into a `LiteralImaginaryUnit` if it is not one already.
Therefore this `__new__` needs overriding

Reply:
So the problem is that `LiteralComplex.__new__` requires the arguments `real` and `imag`?

Reply:
Exactly. If I do:
```
a = LiteralImaginaryUnit()
```
Then it will inherit `__new__` from `LiteralComplex` and complain about missing arguments. The fact that there is an `__init__` has no bearing on the `__new__`